source/blog/openshift-openId-integration-with-engine-sso.html.md

```diff
@@ -0,0 +1,126 @@
+
+## Register a new sso client on ovirty-engine host
+
+Run the new client registration tool ovirt-register-sso-client introduced by patch https://gerrit.ovirt.org/#/c/74212/ to register the new sso client. The tool will prompt the user to enter the client id, location of the client certificate and the callback url prefix. Make note of the client id and client secret generated by the tool. This information is needed in the master-config.yaml file.
+
+## Setup oauthconfig on kibana/elastic search/openshift host 
+
+On kibana/elastic search/openshift host edit the /etc/origin/master/master-config.yaml to setup the oauthconfig as below
+
+```yaml
+oauthConfig:
+  assetPublicURL: https://ocp.origin-14.rmeggins.test.novalocal:8443/console/
+  grantConfig:
+    method: auto
+  identityProviders:
+  - challenge: false
+    login: true
+    mappingMethod: claim
+    name: my_openid_connect
+    provider:
+      apiVersion: v1
+      kind: OpenIDIdentityProvider
+      clientID: <client id specified in previous step>
+      clientSecret: <client id generated in previous step>
+      extraScopes:
+      - ovirt-app-api
+      - ovirt-ext=auth:sequence-priority=~
+      extraAuthorizeParameters:
+        include_granted_scopes: "true"
+claims:
+        id:
+        - custom_id_claim
+        - sub
+        preferredUsername:
+        - preferred_username
+        - email
+        name:
+        - nickname
+        - given_name
+        - name
+        email:
+        - custom_email_claim
+        - email
+      urls:
+        authorize: https://ovirtengine.redhat.com/ovirt-engine/sso/oauth/authorize
+        token: https://ovirtengine.redhat.com/ovirt-engine/sso/oauth/token
+  masterCA: ca-bundle.crt
+  masterPublicURL: https://ocp.origin-14.rmeggins.test.novalocal:8443
+  masterURL: https://host-192-168-78-2.openstacklocal:8443
+  sessionConfig:
+    sessionMaxAgeSeconds: 3600
+    sessionName: ssn
+    sessionSecretsFile: /etc/origin/master/session-secrets.yaml
+  tokenConfig:
+    accessTokenMaxAgeSeconds: 86400
+    authorizeTokenMaxAgeSeconds: 500
+```
+## Restart ovirt-engine
+
+```ssh
+service ovirt-engine restart
+```
+
+## Restart origin-master and origin-node
+
+```ssh
+service origin-master restart
+service origin-node restart
+```
+
+## Configure hostnames
+
+Make sure the hosts are reachable by their hostnames if required add host aliases in /etc/hosts
+
+```config
+10.16.19.48 ocp.origin-14.rmeggins.test.novalocal ocp ocp.origin-14.rmeggins.test kibana.origin-14.rmeggins.test kibana-ops.origin-14.rmeggins.test es.origin-14.rmeggins.test es-ops.origin-14.rmeggins.test
+10.10.116.110 ovirtengine.redhat.com
+```
+
+## Grant permissions
+
+The user needs to be granted permissions manually on the openshift end so the user can view the kibana project
+Accessing https://kibana.origin-14.rmeggins.test should redirect to the engine login page
+Enter credentials and login will redirect user back to kibana
```

Comment:
Don't we need to configure some authorization on Kibana/ES/Openshift side? Everybody is allowed to access? What about admin privileges? I know it's not part of OpenID Connect support in oVirt, but I think it's worth mentioning ...